.github/workflows/ecr-docker-build.yml

```diff
@@ -0,0 +1,50 @@
+name: Docker build and push to ECR
+on:
+  push:
+    branches:
+      - main
+jobs:
+  dockerBuildAndPush:
+    name: Docker build and push
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    environment: test
```

Comment:
Do we need this `environment`. I've had a look at the docs and its use is a little unclear. We don't use it in other workflows though.

Reply:
good spot, killed it